CDCgov/prime-reportstream
# Add default etor filters to UP

This PR adds a framework for having topic-dependent default filters in the UP. I added two maps to `FHIRRouter.kt` which map a topic to a filter, and edited some of the related files to use those maps. I made some adjustments to tests as part of these changes as well.

As a starting point, the etor-ti default quality filter just checks that the message type exists, and the etor-ti default proc mode filter is the same as the existing one for full-elr. That may be as specific as we want to go for now, but we can easily make adjustments in the future, and this provides a pattern that other eventual topics can follow.

Test Steps:
1. Edit the `0149-etor.yml` file to change all instances of `topic: etor-ti` to `topic: full-elr`. Comment out the existing quality filter. Reload those settings by running `./prime multiple-settings set --input ./settings/staging/0149-etor.yml `.
2. Send a lab order by using this postman collection: 
[Temp Collection.postman_collection_json.txt](https://github.com/CDCgov/prime-reportstream/files/11751448/Temp.Collection.postman_collection_json.txt)
3. Query the submission history for the resulting report ID until you see that the report was filtered out by the default quality filter (for the full-elr topic).
4. Edit the `0149-etor.yml` file to change all instances of `topic: full-elr` to `topic: etor-ti`. Leave the existing quality filter commented out. Reload those settings by running `./prime multiple-settings set --input ./settings/staging/0149-etor.yml `.
5. Again send the lab order with the same postman collection as above.
6. Query the submission history for the resulting report ID until you see that the report was delivered to the simulated-lab receiver, meaning it passed the default quality filter for the etor-ti topic.

## Changes
- Changed default quality and proc mode filters to be represented in maps
- Adjusted tests and `isDefaultFilter`/`getQualityFilters`/`getProcessingModeFilter` functions to use the maps

## Checklist

### Testing

## Linked Issues
- Addresses #9464
